### PR TITLE
Update of EntityPickers to support flags which control omit of defaul…

### PR DESCRIPTION
**Pull request by tunkl**
…t namespace in result.

Signed-off-by: Tomáš Tunkl <tomas.tunkl@avast.com>

## Hey, I just made a Pull Request!

Based on #12990 I have updated the EntityPickers so that they support control of default namespace omitting.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder/src/components/fields/EntityPicker/EntityPicker.tsx [hunk 1]
```diff
@@ -57,6 +58,7 @@ export const EntityPicker = (
   } = props;
   const allowedKinds = uiSchema['ui:options']?.allowedKinds;
   const defaultKind = uiSchema['ui:options']?.defaultKind;
+  const defaultNamespace = uiSchema['ui:options']?.defaultNamespace;
```
Comment (freben) [inline plugins/scaffolder/src/components/fields/EntityPicker/EntityPicker.tsx:61]: How did you plan to represent the "false" case? The code here seems to allow undefined, but the code further down does `?? ''` which probably is worse.

Comment (tunkl) [reply plugins/scaffolder/src/components/fields/EntityPicker/EntityPicker.tsx:61]: I admit that I am little bit lost. The ?? '' is the only way how I was able to make it work. If I remove ?? '' which I would expect to be the right way, is incompatible.

File: plugins/scaffolder/src/components/fields/OwnerPicker/OwnerPicker.tsx [hunk 2]
```diff
@@ -53,6 +54,7 @@ export const OwnerPicker = (
       defaultKind: 'Group',
       allowArbitraryValues:
         uiSchema['ui:options']?.allowArbitraryValues ?? true,
+      defaultNamespace: uiSchema['ui:options']?.defaultNamespace ?? '',
```
Comment (freben) [inline plugins/scaffolder/src/components/fields/OwnerPicker/OwnerPicker.tsx:57]: You could do

```ts
const defaultNamespace = uiSchema['ui:options']?.defaultNamespace;

const ownerUiSchema = {
  ...uiSchema,
  'ui:options': {
    // ... other options
    ...( defaultNamespace !== undefined ? { defaultNamespace } : {})
  }
```

**Review by freben — COMMENTED** [review]
OK, just one final comment I think.

File: plugins/scaffolder/src/components/fields/OwnerPicker/OwnerPicker.tsx (hunk 2 above)
Comment (tunkl) [reply plugins/scaffolder/src/components/fields/OwnerPicker/OwnerPicker.tsx:57]: Sorry, but you mean something like this?
```
const defaultNamespace = uiSchema['ui:options']?.defaultNamespace;

  const ownerUiSchema = {
    ...uiSchema,
    'ui:options': {
      allowedKinds: (uiSchema['ui:options']?.allowedKinds || [
        'Group',
        'User',
      ]) as string[],
      defaultKind: 'Group',
      allowArbitraryValues:
        uiSchema['ui:options']?.allowArbitraryValues ?? true,
      defaultNamespace: ( defaultNamespace !== undefined ? { defaultNamespace } : {})
    },
  };
```

that doesn't work :(

Comment (freben) [reply plugins/scaffolder/src/components/fields/OwnerPicker/OwnerPicker.tsx:57]: No I think I meant precisely what I wrote, if my in-brain compiler works the way it should :)

Screenshot 2022-09-21 at 17 05 19

Comment (tunkl) [reply plugins/scaffolder/src/components/fields/OwnerPicker/OwnerPicker.tsx:57]: Ok this works. I have to learn about typescript a lot :-D

**Review by freben — APPROVED** [review]

**Pull request merged**